contrib/clojure-package/README.md

```diff
@@ -80,75 +83,66 @@ wget https://archive.archlinux.org/packages/c/cuda/cuda-9.0.176-4-x86_64.pkg.tar
 
+- `git clone --recursive https://github.com/apache/incubator-mxnet.git ~/mxnet`
+- `cd mxnet`
+- `git tag —list` (find the tag that corresponds the version of the latest Scala jar)
+- `git checkout tags/<tag_name> -b <branch_name>`
+- `cd contrib/clojure`
+- edit `project.clj` to include your Scala jar from maven. It should match your system. Example `[org.apache.mxnet/mxnet-full_2.11-linux-x86_64-cpu "x.y.z”]`
+- run `lein test`. All the tests should run without an error
+- At this point you can do `lein install` to build and install the Clojure jar locally. Now, you can run the examples by doing `cd examples/imclassification` and then `lein run`  or `lein run :cpu 2` (for 2 cpus) (for gpu `lein run :gpu`)
 
-### Build from MXNET Source
+You also might need dependencies installed based on your system. Please see the dependency section above in [Getting Started with the Clojure Jars on Maven](#getting-started-with-the-clojure-jars-on-maven))
 
-First, ensure you have JDK 8 on your system. Later versions may produce cryptic build errors mentioning `scala.reflect.internal.MissingRequirementError`. 
 
-Checkout the latest SHA from the main package:
+### Getting Started with MXNet project with Building from Source
 
-`git clone --recursive https://github.com/apache/incubator-mxnet.git ~/mxnet`
-`cd ~/mxnet`
+This option will build the C lib locally and then the Scala jars as well. You can use this option when you are developing changes for the project.
 
-If you need to checkout a particular release you can do it with:
+* For your dependencies look at https://mxnet.incubator.apache.org/install/osx_setup.html or https://mx	net.incubator.apache.org/install/ubuntu_setup.html
+	about _Prepare Environment for GPU Installation_
+	and	 _Install MXNet dependencies_
+* Also, ensure you have JDK 8 on your system. Later versions may produce cryptic build errors mentioning `scala.reflect.internal.MissingRequirementError`. 
+* Checkout the latest SHA from the main package:
+* `git clone --recursive https://github.com/apache/incubator-mxnet.git ~/mxnet`
+* `cd ~/mxnet`
+* `git tag —list` (find the tag that corresponds the version of the latest Scala jar)
+* `git checkout tags/<tag_name> -b <branch_name>` To checkout a particular release
+* `git submodule update --init --recursive` - To updated all the submodules to the commit
+  * Sometimes it is useful to use this script to clean hard, (remove any extra files and reset everything), using this [gist](https://gist.github.com/nick	toumpelis/11214362)
+* Go here to do the base package installation https://mxnet.incubator.apache.org/install/index.html.
+  * This will create an `lib/libmxnet.so` file that is the c api we need for the next step.
```

Comment:
```suggestion
  * This will create an `lib/libmxnet.so` file that MXNet shared library that we need for the next step.
```
There's debate about how much to expose re: the C API since it really isn't officially supported. So maybe not shine a light on it...